### ISSUE TITLE
No validation for controller prefixes and controller methods 

### ISSUE THREAD

**Issue by sztadii** [nestjs/nest#4813]
## Current behavior
Firstly I need to say, great job :) The framework is really good, but with few minor bugs to fix.
So currently we are able to build and start the project with the same prefixes for two different controllers and a single controller is able to have the same endpoint path. I think additional validation should improve the framework development experience.

![Screenshot 2020-05-16 at 22 01 40](https://user-images.githubusercontent.com/29600182/82127063-fc900280-97c1-11ea-9602-5baa4a92017d.png)

## Possible Solution
I think method `RoutesResolver.registerRouters` should handle this case and throw an error if there are no unique paths/endpoints

## Environment
Operation system: macOS Catalina 10.15.4
Node version: 12.14.1

**Comment by kamilmysliwiec** [comment]
Thanks @sztadii!

> So currently we are able to build and start the project with the same prefixes for two different controllers

There's nothing wrong in having 2 controller classes with the same prefix (for example, if you have a lot of endpoints, you can split them into Read/Write controllers).

> a single controller is able to have the same endpoint path 

This scenario isn't common, but since we want to remain compatible with "express", you can bind multiple handlers to the same endpoint. For example, if you have 2 handlers bound to the same endpoint, you can inject the `@Next()` function in the first one, and once you call this function, Nest will execute the second handler. This isn't recommended since it heavily complicates the request-response flow, but still, we must support this edge case to preserve the compatibility. 

TL;DR; no validation is intentional.
